mgmt/src/grpc/server.rs

```diff
@@ -0,0 +1,1617 @@
+    ) -> Result<VpcManifest, String> {
+        // Create a new VPC manifest with the VPC name
+        let mut manifest = VpcManifest::new(&entry.vpc);
+
+        // Process each expose rule
+        for expose_grpc in &entry.expose {
+            let expose = self.convert_expose_from_grpc(expose_grpc)?;
+            manifest.add_expose(expose).map_err(|e| {
+                format!(
+                    "Failed to add expose to manifest for VPC {}: {}",
+                    entry.vpc, e
+                )
+            })?;
+        }
+
+        Ok(manifest)
+    }
+
+    /// Convert gRPC Expose to VpcExpose
+    fn convert_expose_from_grpc(
+        &self,
+        expose: &gateway_config::Expose,
+    ) -> Result<VpcExpose, String> {
+        // Start with an empty expose
+        let mut vpc_expose = VpcExpose::empty();
+
+        // Process IP rules
+        for ip in &expose.ips {
+            if let Some(rule) = &ip.rule {
+                match rule {
+                    gateway_config::config::peering_i_ps::Rule::Cidr(cidr) => {
+                        // Parse CIDR into IP and netmask
+                        let (ip_str, netmask) = self.parse_cidr(cidr)?;
+                        // Add as an include rule
+                        vpc_expose = vpc_expose.ip(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                    gateway_config::config::peering_i_ps::Rule::Not(not) => {
+                        // Parse CIDR into IP and netmask for exclude rule
+                        let (ip_str, netmask) = self.parse_cidr(not)?;
+                        // Add as an exclude rule
+                        vpc_expose = vpc_expose.not(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                }
+            } else {
+                return Err("PeeringIPs must have either cidr or not field set".to_string());
+            }
+        }
+
+        // Process AS rules
+        for as_rule in &expose.r#as {
+            if let Some(rule) = &as_rule.rule {
+                match rule {
+                    gateway_config::config::peering_as::Rule::Cidr(cidr) => {
+                        // Parse CIDR into IP and netmask
+                        let (ip_str, netmask) = self.parse_cidr(cidr)?;
+                        // Add as an include rule for AS
+                        vpc_expose = vpc_expose.as_range(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                    gateway_config::config::peering_as::Rule::Not(not) => {
+                        // Parse CIDR into IP and netmask for exclude rule
+                        let (ip_str, netmask) = self.parse_cidr(not)?;
+                        // Add as an exclude rule for AS
+                        vpc_expose = vpc_expose.not_as(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                }
+            } else {
+                return Err("PeeringAs must have either cidr or not field set".to_string());
```

Comment:
```suggestion
                return Err("PeeringAs must have either 'cidr' or 'not' field set".to_string());
```

`not` is confusing otherwise. Also applies elsewhere in the file (such as in the previous `for` loop).

Reply:
it's what we have in our API... I will rename it as `ip_exclude` in a code, but for comment - we need to align with API implementation

Reply:
I don't mean changing the names, just adding quotes. Is that introducing a contradiction with the API?

Reply:
Totally no, we just highlighting exact API fields. Thanks for hit with quotes, added in the code